src/backend/marsha/core/serializers/file.py

```diff
@@ -173,3 +175,32 @@ class InitiateUploadSerializer(serializers.Serializer):
 
     filename = serializers.CharField()
     mimetype = serializers.CharField(allow_blank=True)
+
+
+class SharedLiveMediaInitiateUploadSerializer(InitiateUploadSerializer):
+    """An initiate-upload serializer dedicated to shared live media."""
+
+    def validate(self, attrs):
+        """Validate if the mimetype is allowed or not."""
+        # mimetype is provided, we directly check it
+        if attrs["mimetype"] != "":
+            if attrs["mimetype"] not in settings.ALLOWED_SHARED_LIVE_MEDIA_MIME_TYPES:
+                raise serializers.ValidationError(
+                    {"mimetype": f"{attrs['mimetype']} is not a supported mimetype"}
+                )
+            attrs["extension"] = mimetypes.guess_extension(attrs["mimetype"])
+
+        # mimetype is not provided, we have to guess it from the extension
+        if attrs["mimetype"] == "":
+            mimetypes.init()
+            extension = splitext(attrs["filename"])[1]
+            mimetype = mimetypes.types_map.get(extension)
+            if mimetype not in settings.ALLOWED_SHARED_LIVE_MEDIA_MIME_TYPES:
+                raise serializers.ValidationError(
+                    {"mimetype": "mimetype not guessable"}
```

Comment:
```suggestion
                    {"mimetype": f"{mimetype} is not a supported mimetype"}
```
Shouldn't we raise this error ?

Reply:
We guess it so I don't know if we can say that this exact `mimetype` is wrong. IMO if the client wants to fix the error he has to provide a mimetype otherwise it's subject to errors.

Reply:
Seems legit!